libraries/entities-renderer/src/RenderableEntityItem.cpp

```diff
@@ -491,6 +495,176 @@ void EntityRenderer::removeMaterial(graphics::MaterialPointer material, const st
     emit requestRenderUpdate();
 }
 
+EntityRenderer::Pipeline EntityRenderer::getPipelineType(const graphics::MultiMaterial& materials) {
```

Comment:
all of the changes in this file are moved from RenderableShapeEntityItem more or less without change, so that this all can be reused for other entities that support materials